plutus-tx-plugin/test/Blueprint/Tests/Lib.hs

```diff
@@ -125,17 +131,17 @@ newtype Param2a = MkParam2a Bool
   deriving stock (Generic)
   deriving anyclass (AsDefinitionId)
 
-$(PlutusTx.makeLift ''Param2a)
+$(makeLift ''Param2a)
 $(makeIsDataSchemaIndexed ''Param2a [('MkParam2a, 0)])
 
 newtype Param2b = MkParam2b Bool
   deriving stock (Generic)
   deriving anyclass (AsDefinitionId)
 
-$(PlutusTx.makeLift ''Param2b)
+$(makeLift ''Param2b)
 $(makeIsDataSchemaIndexed ''Param2b [('MkParam2b, 0)])
 
-type Datum2 = Integer
+$(asData datum2)
```

Comment:
Could we define a schema instance here in terms of the one for the non-asData type definition? Then you could keep things local.

Reply:
i.e. import it qualified, then do
```
instance ... where
   schemaFunctionWhatever = BlueprintTypes.schemFunctionWhatever
```

Reply:
The `HasSchema` instance is codegenerated with TH; As it is implemented atm its not possible to import the schema definition only `BlueprintTypes.schemFunctionWhatever`.

Reply:
I don't understand. Yes, the instance that you generate in `Blueprint.hs` is generated. That does not mean that you can't _use_ that instance to define the instance for `Datum2` in `Lib.hs`?

Reply:
The module `Blueprint.Tests.Lib.AsData.Blueprint` uses TH to generate
```
instance (HasSchemaDefinition Integer referencedTypes,
          HasSchemaDefinition Bool referencedTypes) =>
         HasSchema Datum2 referencedTypes where
  {-# INLINABLE schema #-}
  schema
    = SchemaConstructor
        (MkSchemaInfo Nothing Nothing Nothing)
        (MkConstructorSchema
           0
           [definitionRef @Integer @referencedTypes,
            definitionRef @Bool @referencedTypes])
```
and you propose to use this implementation of `schema` to define another instance for the same type in the `Blueprint.Tests.Lib`  ?

Please help me better understand the [locality] problem you're trying to solve and the solution you're suggesting.

Reply:
So at the moment you have to import the "schema-only" type here: https://github.com/IntersectMBO/plutus/pull/5837/files#diff-310cdc5eee57d553b6ea5fdc8b1590aa1333b9c6d8ef15577b51522517953c77R13

So that you can refer to that type instead of the actual type you are actually using. I'm suggesting to avoid this by giving the _actual_ `Datum2` type a `HasSchema` definition like so:

```
import Blueprint.Tests.Lib.AsData.Blueprint qualified as SchemaTypes

instance (HasSchemaDefinition Integer referencedTypes,
          HasSchemaDefinition Bool referencedTypes) =>
         HasSchema Datum2 referencedTypes where

  schema = schema @SchemaTypes.Datum2
```

now you can use the _actual_ `Datum2` everywhere because it has the right instnace.

Reply:
Now I see what you meant! Done here https://github.com/IntersectMBO/plutus/pull/5877